nntrainer/models/neuralnet.cpp

```diff
@@ -675,11 +676,20 @@ int NeuralNetwork::train_run() {
     stat.loss += getLoss();
   };
 
-  auto eval_epoch_end = [this, batch_size](RunStats &stat, DataBuffer &buffer) {
+  auto eval_epoch_end = [this, batch_size, max_acc = 0.0f](
+                          RunStats &stat, DataBuffer &buffer) mutable {
     stat.loss /= static_cast<float>(stat.num_iterations);
     stat.accuracy = stat.num_correct_predictions /
                     static_cast<float>(stat.num_iterations * batch_size) *
                     100.0f;
+
+    if (stat.accuracy >= max_acc) {
```

Comment:
`if (stat.accuracy > max_acc || (stat.accuracy > max_acc && stat.loss < min_loss)) {`
and cache a min_loss as well.

Reply:
updated :)